vyper.asciidoc

```diff
@@ -93,11 +96,83 @@ Inline assembly:: Inline assembly provides developers with an opportunity to acc
 3 0x80 mload add 0x80 mstore
 ----
 
-As mentioned previously, Vyper strives to provide developers and code auditors with the most human readable code. whilst inline assembly can provide powerful fine-grained control, it is not supported by the Vyper programming language.
+As mentioned previously, Vyper strives to provide developers and code auditors with the most human readable code. Whilst inline assembly can provide powerful fine-grained control, it is not supported by the Vyper programming language.
+
+
+Function overloading:: Function overloading allows developers to write multiple functions of the same name. Each of the overloaded functions, which contain the same name, are uniquely identified by their argument signatures. Take the following two functions for example.
+
+[source,javascript]
+----
+function f(uint _in) public pure returns (uint out) {
+    out = 1;
+}
+
+function f(uint _in, bytes32 _key) public pure returns (uint out) {
+    out = 2;
+}
+----
+The first function (named f) accepts an input argument of type uint, however the second function (also named f) accepts two arguments, one of type uint and one of type bytes32. Having multiple function definitions with the same name and different argument options could cause confusion and it is primarily for this reason that Vyper does not support function overloading.
+
```

Comment:
However, Vyper will introduce support for default parameters in functions. This shall allow basic "overloading" replacement as the ABI will generate different functions for each version of the default parameters. This support will allow better support for Solidity standards (like ERC721) which contains some basic overloading.